docs/current_docs/developer-guide/go/snippets/services/return-value/main.go

```diff
@@ -0,0 +1,31 @@
+package main
+
+import (
+	"context"
+)
+
+type MyModule struct{}
+
+// starts and returns an HTTP service
+func (m *MyModule) HttpService() *Service {
+	return dag.Container().
+		From("python").
+		WithDirectory("/srv", dag.Directory().WithNewFile("index.html", "Hello, world!")).
+		WithWorkdir("/srv").
```

Comment:
I'd serve a directory coming as a function argument here:
```go
		WithWorkdir("/srv").
		WithMountedDirectory("", src).
```

Otherwise, you can also add files to Container directly:
```go
		WithWorkdir("/srv").
		WithNewFile("index.html", ContainerWithNewFileOpts{
			Contents: "Hello, world!",
		}).
```

Reply:
Updated